### PR TITLE
fix: completion state for connectivity test, audio test ui

### PR DESCRIPTION
**Pull request by KaustubhKumar05**
### Fixes:

https://100ms.atlassian.net/browse/WEB-2884 (Connection quality completion state)
https://100ms.atlassian.net/browse/WEB-2887 (Output selection for Safari)
https://100ms.atlassian.net/browse/WEB-2891 (Update audio test UI)
https://100ms.atlassian.net/browse/WEB-2893 (Output selection for mweb)

### Before:

image

image

### After:

Going back to in progress if the test is run again and on completion, the state is updated

image

### CODE REVIEW

File: packages/roomkit-react/src/Diagnostics/AudioTest.tsx [hunk 1]
```diff
@@ -84,37 +112,23 @@ const SpeakerTest = () => {
   const actions = useHMSActions();
   const devices = useHMSStore(selectDevices);
   const { audioOutputDeviceId } = useHMSStore(selectLocalMediaSettings);
-  const { playing, setPlaying, audioRef } = useAudioOutputTest({ deviceId: audioOutputDeviceId || 'default' });
+  const isMobile = useMedia(cssConfig.media.md);
+
+  if (isMobile || isSafari) {
```
Comment (raviteja83) [inline packages/roomkit-react/src/Diagnostics/AudioTest.tsx:117]: in safari do we get the output device in device list? we can check the length of audioOutput. That would be a better check instead of isMobile and isSafari. covers both.

File: packages/roomkit-react/src/Diagnostics/Diagnostics.tsx [hunk 2]
```diff
@@ -77,23 +77,20 @@ const DiagnosticsStep = () => {
...
+  const { activeStep, connectivityTested } = useContext(DiagnosticsContext);
 
   return (
     <Box css={{ w: '25%', '@lg': { display: 'none' } }}>
       {Object.keys(DiagnosticsSteps).map(key => {
         const keys = Object.keys(DiagnosticsSteps);
         const activeStepIndex = keys.indexOf(activeStep);
         const keyIndex = keys.indexOf(key);
-        const isStepCompleted = activeStepIndex > keyIndex;
+        const isStepCompleted = activeStepIndex > keyIndex || (activeStep === 'connectivity' && connectivityTested);
```
Comment (raviteja83) [inline packages/roomkit-react/src/Diagnostics/Diagnostics.tsx:88]: possible to have a state in the context.

[{ 
  step: 'name',
 completed: boolean,
active: boolean
},
{
step: 'name',
 completed: boolean,
active: boolean
}

]

File: packages/roomkit-react/src/Diagnostics/AudioTest.tsx (hunk 1 above)
Comment (KaustubhKumar05) [reply packages/roomkit-react/src/Diagnostics/AudioTest.tsx:117]: Ack, will check this

File: packages/roomkit-react/src/Diagnostics/Diagnostics.tsx (hunk 2 above)
Comment (KaustubhKumar05) [reply packages/roomkit-react/src/Diagnostics/Diagnostics.tsx:88]: This is only needed for the final step since it can be re-run. For the others the existing logic of checking the active step index is sufficient, didn't want to complicate it. Can update if this has some edge cases

Comment (eswarclynn) [reply packages/roomkit-react/src/Diagnostics/Diagnostics.tsx:88]: This what what I was thinking earlier, we also need an error property here. I can pick this up next

File: packages/roomkit-react/src/Diagnostics/AudioTest.tsx (hunk 1 above)
Comment (KaustubhKumar05) [reply packages/roomkit-react/src/Diagnostics/AudioTest.tsx]: This works, updated the PR

File: packages/roomkit-react/src/Diagnostics/Diagnostics.tsx (hunk 2 above)
Comment (raviteja83) [reply packages/roomkit-react/src/Diagnostics/Diagnostics.tsx:88]: Let us keep it as enahancements then.

**Review by raviteja83 — APPROVED** [review]

File: packages/roomkit-react/src/Diagnostics/Diagnostics.tsx [hunk 3]
```diff
@@ -77,23 +77,20 @@ const DiagnosticsStep = () => {
...
 
         let color = '$on_primary_low';
         if (isStepCompleted) {
           color = '$primary_bright';
         }
-        if (activeStep === key) {
-          color = '$on_primary_high';
-        }
```
Comment (eswarclynn) [inline packages/roomkit-react/src/Diagnostics/Diagnostics.tsx:96]: Why was this removed @KaustubhKumar05?
